server/api/api/endpoints/functions.py

```diff
@@ -185,6 +187,26 @@ async def delete_function(
             await run_in_threadpool(
                 get_scheduler().delete_schedule, db_session, project, name
             )
+
+    # getting all function tags
+    functions = await run_in_threadpool(
+        server.api.crud.Functions().list_functions, db_session, project, name
+    )
+    if len(functions) > 0:
+        if functions[0].get("kind") in RuntimeKinds.nuclio_runtimes():
+            async with server.api.utils.clients.async_nuclio.Client(
+                auth_info
+            ) as client:
+                tasks = []
+                for function in functions:
+                    nuclio_name = mlrun.runtimes.nuclio.function.get_fullname(
+                        name, project, function.get("metadata", {}).get("tag")
+                    )
+                    tasks.append(
+                        client.delete_function(name=nuclio_name, project_name=project)
+                    )
+                await asyncio.gather(*tasks)
```

Comment:
few things here are not very clear
1. the heuristic if checking the first function is nuclio and assuming rest are nuclio - perhaps worth a comment why do oyu assume it
2. gather on tasks - if you have 500 functions, do you want to send 500 requests in parallel? why not group those requests by 10ens and send them all (remember not to fail on first but send all requests, log what fails and if one failed, then at the end - fail the operation
3. will you wait for functions deletion? Id assume that is a long-process - thus, needs to be a background job.
4. as stated on (2) - gather will fail if one deletion has failed. I suggest to wait for all before bailing

---
since (3) requires a change on response, I suggest moving this operation to functions_v2.py and have this function return background job
then you update 
httpdb (sdk) to expect background job (and use v2)
open ticket for ui to use /v2/... endpoint and expect background job

Reply:
@liranbg reworked